pkg/controller/disruption/disruption.go

```diff
@@ -596,13 +588,13 @@ func (dc *DisruptionController) getExpectedScale(pdb *policy.PodDisruptionBudget
 	for _, pod := range pods {
 		controllerCount := 0
 		for _, finder := range dc.finders() {
-			var controllers []controllerAndScale
-			controllers, err = finder(pod)
+			var controllerNScale *controllerAndScale
+			controllerNScale, err = finder(pod)
 			if err != nil {
 				return
 			}
-			for _, controller := range controllers {
-				controllerScale[controller.UID] = controller.scale
+			if controllerNScale != nil {
```

Comment:
Since we use ControllerRef now, it's not possible for more than one finder to return a match. So you can change `controllerCount int` to `foundController bool` and add a `break` after you set `foundController = true` in this block. You can also get rid of the check for >1 controller on a single pod.